src/server/common_services/auth_server.js

```diff
@@ -21,6 +22,14 @@ const config = require('../../../config');
 const s3_bucket_policy_utils = require('../../endpoint/s3/s3_bucket_policy_utils');
 
 
+
+function compare_password_hash(password, target_account) {
+    if (bcrypt.compare(password.unwrap(), target_account.password.unwrap())) {
+        return true;
+    }
```

Comment:
What is this supposed to do - are we still allowing previous bcyrpt?
This needs to be at least explained in a code comment.